### ISSUE TITLE
Point-wise weighted loss

### ISSUE THREAD

**Issue by jcribeiro98** [open-mmlab/mmclassification#515]
Hi! We are looking for a way to implement a point wise weighted cross entropy loss, similar to the one that you can find [here](https://mmclassification.readthedocs.io/en/latest/api.html#mmcls.models.losses.cross_entropy), but instead of using class-wise weights, I want to use a array of floats for each point used in the loss.

We looked in the docs for something like this, but we only found that class-wise loss. 

Thanks!

**Comment by Ezra-Yu** [comment]
Do you mean element-wise loss?

You could use `weight` in [forward function](https://github.com/open-mmlab/mmclassification/blob/29f512156c526df6748da97d1a8b55879cdabf9e/mmcls/models/losses/cross_entropy_loss.py#L168). Some cases can be seen in [tests](https://github.com/open-mmlab/mmclassification/blob/29f512156c526df6748da97d1a8b55879cdabf9e/tests/test_metrics/test_losses.py#L73)

**Comment by jcribeiro98** [comment]
Yes, but the thing is that I want to update those weights each iteration.

Those weights are the result of another nn, I have something like this:

```
for batch in batches:
    loss_element_wise_weight = torch.rand(batch_size)
    train_stage(batch)
```

**Comment by mzr1996** [comment]
Hello, if you want to update these weights per iteration, I think you can use a hook to implement it.
for example, add a `element_weights` buffer in your network head, pass it to the `weight` argument when forward, and update this buffer in the hook after every iteration.

**Comment by Ezra-Yu** [comment]
We find that there is no args or kwargs for `weight` in heads.froward_train such as [LinearClsHead.froward_train](https://github.com/open-mmlab/mmclassification/blob/dc35eb62b61ab17decedb93c4d83ec6da5f93eff/mmcls/models/heads/linear_head.py#L49), which may make users hard to use `weight` in BCE loss. We are going to add that.
